catroid/src/main/java/org/catrobat/catroid/utils/CrashReporter.java

```diff
@@ -0,0 +1,134 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.utils;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.support.annotation.VisibleForTesting;
+import android.util.Log;
+
+import com.crashlytics.android.Crashlytics;
+import com.google.gson.Gson;
+
+import org.catrobat.catroid.BuildConfig;
+import org.catrobat.catroid.ui.MainMenuActivity;
+import org.catrobat.catroid.ui.SettingsActivity;
+
+import io.fabric.sdk.android.Fabric;
+
+public final class CrashReporter {
+
+	private static final String TAG = CrashReporter.class.getSimpleName();
+
+	public static final String RECOVERED_FROM_CRASH = "RECOVERED_FROM_CRASH";
+	public static final String EXCEPTION_FOR_REPORT = "EXCEPTION_FOR_REPORT";
+
+	private static SharedPreferences preferences;
+	private static boolean isCrashReportEnabled = BuildConfig.CRASHLYTICS_CRASH_REPORT_ENABLED;
+
+	private CrashReporter() {
+	}
+
+	public static boolean initialize(Context context) {
+
+		preferences = PreferenceManager.getDefaultSharedPreferences(context);
+
+		if (isReportingEnabled()) {
+			Fabric.with(context, new Crashlytics());
+			Log.d(TAG, "INITIALIZED!");
+			return true;
+		}
+		Log.d(TAG, "INITIALIZATION FAILED! [ Anonymous Report : " + isAnonymousReportEnabled() + " Crash Report : " + isCrashReportEnabled + " ]");
+		return false;
+	}
+
+	private static boolean isAnonymousReportEnabled() {
+		return preferences != null && preferences.getBoolean(SettingsActivity.SETTINGS_CRASH_REPORTS, false);
+	}
+
+	private static boolean isRecoveredFromCrash() {
+		return preferences != null && preferences.getBoolean(RECOVERED_FROM_CRASH, false);
+	}
+
+	private static boolean isReportingEnabled() {
+		return isAnonymousReportEnabled() && isCrashReportEnabled;
+	}
+
+	@VisibleForTesting
+	public static void setIsCrashReportEnabled(boolean isEnabled) {
+		isCrashReportEnabled = isEnabled;
+	}
+
+	public static boolean logException(Throwable t) {
+
+		if (isReportingEnabled()) {
+			Crashlytics.logException(t);
+			return true;
+		}
+
+		return false;
+	}
+
+	public static boolean checkIfCrashRecoveryAndFinishActivity(final Activity context) {
+		Log.d(TAG, "AFTER_EXCEPTION : checkIfCrashRecoveryAndFinishActivity()");
+		if (isRecoveredFromCrash()) {
+			if (isReportingEnabled()) {
+				sendUnhandledCaughtException();
+			}
+			if (!(context instanceof MainMenuActivity)) {
+				context.finish();
+			} else {
+				preferences.edit().putBoolean(RECOVERED_FROM_CRASH, false).commit();
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public static void sendUnhandledCaughtException() {
+		Log.d(TAG, "AFTER_EXCEPTION : sendCaughtException()");
+		Gson gson = new Gson();
+		String json = preferences.getString(EXCEPTION_FOR_REPORT, "");
+		Throwable exception = gson.fromJson(json, Throwable.class);
```

Comment:
I think this issue is supposed to be handle [here #2366](https://github.com/Catrobat/Catroid/pull/2366). I will be updating it soon.